libraries/ESP8266WiFi/src/include/ClientContext.h

```diff
@@ -325,6 +328,26 @@ class ClientContext
         return _write_from_source(new BufferedStreamDataSource<ProgmemStream>(stream, size));
     }
 
+    void keepAlive (uint16_t idle_sec = TCP_DEFAULT_KEEPALIVE_IDLE_SEC, uint16_t intv_sec = TCP_DEFAULT_KEEPALIVE_INTERVAL_SEC, uint8_t count = TCP_DEFAULT_KEEPALIVE_COUNT)
+    {
+        if (idle_sec && intv_sec && count) {
+            _pcb->so_options |= SOF_KEEPALIVE;
+            _pcb->keep_idle = (uint32_t)1000 * idle_sec;
+            _pcb->keep_intvl = (uint32_t)1000 * intv_sec;
+            _pcb->keep_cnt = count;
+        }
+        else
+            _pcb->so_options &= ~SOF_KEEPALIVE;
+    }
+
+    void getKeepAlive (uint16_t* idle_sec, uint16_t* intv_sec, uint8_t* count)
```

Comment:
Maybe use of references instead of pointers would be better here?

Reply:
This way with pointers let the user choose which value to get, like with ```gettimeofday(&tv, NULL)``` we never want the second one. This can't be done with references
I put this ```getKeepAlive()``` method mainly for the user to be informed if keep-alive is activated through ```WiFiClient::``` since ```ClientContext``` is not accessible.
I'm open to any other method than this one.

Reply:
I missed your original intent. In that case, I think I'd prefer three inline methods, one for each value, each returning the requested value, plus a 4th inline method that just returns the enabled flag as a bool, something along the lines of:
getKeepAliveCount()
getKeepAliveInterval()
getKeepAliveIdle()
isKeepAliveEnabled()

Reply:
Body could be like this:
return isKeepAliveEnabled() ? _pcb-> blah : 0;

Reply:
keepalive-api is updated, PR squashed-rebased